### DISCUSSION TITLE
Dynamic timeline variables

### DISCUSSION THREAD

**Discussion by MilenR** [jspsych/jsPsych#2302] [Q&A] [answered]
Hi!
I'm using Jspsych and Pavlovia. Right now I want to do the following:

1. The user enters a number
2.  According that number a different variables set is chosen and determines the experiment.

Is it possible to do this? I've tried creating a variable 'user_id', the user enters it with the survey_text plugin and according that the set changes. But no matter what value the user enters, the default set is always chosen.
I'm trying to figure out an alternative so we don't have to create n experiments in Pavlovia (whose only difference is only the variables).
Thanks so much for the help!

**Comment by jodeleeuw** [comment, marked as answer]
Hi @MilenR 

Currently `timeline_variables` cannot be a function. It is something that we would like to add support for soon.

A less elegant solution with the current version of jsPsych would be to use the [conditional function](https://www.jspsych.org/7.0/overview/timeline/#conditional-timelines) on a timeline. You could set up *n* timelines, each with a `conditional_function`, so that only one of the timelines runs based on the answer to the `survey-text` trial.

**Comment by MilenR** [comment]
Hi @jodeleeuw 
Just read the documentation and we believe the conditional function will work for us. 
Thanks so much for the help!
